coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -61,13 +69,46 @@ func (bp *BatchProverTask) Assign(ctx *gin.Context, getTaskParameter *coordinato
 		return nil, fmt.Errorf("check prover task parameter failed, error:%w", err)
 	}
 
+	var hardForkNumber uint64
+	if getTaskParameter.HardForkName != "" {
+		var exist bool
+		hardForkNumber, exist = bp.nameForkMap[getTaskParameter.HardForkName]
+		if !exist {
+			log.Debug("hard fork prover get empty batch because of the hard fork name don't exist", "height",
+				getTaskParameter.ProverHeight, "fork name", getTaskParameter.HardForkName)
+			return nil, nil
+		}
+	}
+
+	if getTaskParameter.HardForkName == "" {
+		hardForkNumber = bp.maxForkNumber - 1
+	}
+
+	// if the hard fork number set, rollup relayer must generate the chunk from hard fork number,
+	// so the hard fork chunk's start_block_number must be ForkBlockNumber
+	hardForkStartChunk, getStartChunkErr := bp.chunkOrm.GetChunkByStartBlockNumber(ctx, bp.maxForkNumber)
```

Comment:
This works but it seems overcomplicated to me.

To me, the logical approach would be:
- Based on `hardForkNumber`, get hard fork block range. The results is e.g. `[0, 99]` and `[100, MAX]`.
- In db chunk query, add condition on block range

Reply:
`bp.batchOrm.GetAssignedBatch` query the batch table, only have `start_chunk_index` and `end_chunk_index` in the batch table. So if batch want know the `start_block_number` of batch, batch sql need inner join with chunk.

So I split the inner join sql.

Reply:
What I don't like about this approach are:
- Special handing for the "last fork". If we spin up a test network and set forks to `100` and `200` it will only care about the last one.
- Passing `isFork` to the DB queries. I feel like the orm layer does not need to be aware of forks.

Reply:
make sense

Reply:
fixed [5da9b94](https://github.com/scroll-tech/scroll/pull/1182/commits/5da9b94d1bd0e1a3029a5f0f00b37d4d7b74b3be)